packages/core/src/lib/models/link.model.ts

```diff
@@ -48,17 +48,17 @@ export class LinkModel extends BaseModel<DiagramModel> {
     );
   }
 
-  setExtras<E>(extras: Partial<E>) {
-    this.extras$.set(extras);
+  setExtras<Extras>(extras: Partial<Extras>) {
```

Comment:
I think it's worth adding an `upsert` method for the extras that will merge the current state with the changes, instead of doing that in the `set` method